src/game.cpp

```diff
@@ -10583,6 +10583,9 @@ point game::update_map( int &x, int &y )
         shift.y++;
     }
 
+    // Update what parts of the world map we can see
+    update_overmap_seen();
```

Comment:
I think you probably need to put this back where it was and add an additional call right before the early return on 10593.  Pretty sure it needs to happen after any potential map shift